consensus/core/src/core.rs

```diff
@@ -1588,39 +1656,59 @@ mod test {
         }
 
         // Check commits have been persisted to store
-        let last_commit = store
+        let last_commit = core_fixture
+            .store
             .read_last_commit()
             .unwrap()
             .expect("last commit should be set");
         // There are 8 leader rounds with rounds completed up to and including
         // round 10. However because there were no blocks produced for authority 3
         // 2 leader rounds will be skipped.
         assert_eq!(last_commit.index(), 6);
-        let all_stored_commits = store.scan_commits((0..CommitIndex::MAX).into()).unwrap();
+        let all_stored_commits = core_fixture
+            .store
+            .scan_commits((0..CommitIndex::MAX).into())
+            .unwrap();
         assert_eq!(all_stored_commits.len(), 6);
     }
 
     /// Creates cores for the specified number of authorities for their corresponding stakes. The method returns the
     /// cores and their respective signal receivers are returned in `AuthorityIndex` order asc.
-    // TODO: return a test fixture instead.
-    fn create_cores(
-        context: Context,
-        authorities: Vec<Stake>,
-    ) -> Vec<(
-        Core,
-        CoreSignalsReceivers,
-        broadcast::Receiver<VerifiedBlock>,
-        UnboundedReceiver<CommittedSubDag>,
-        Arc<impl Store>,
-    )> {
+    pub(crate) fn create_cores(context: Context, authorities: Vec<Stake>) -> Vec<CoreTextFixture> {
         let mut cores = Vec::new();
 
         for index in 0..authorities.len() {
+            let own_index = AuthorityIndex::new_for_test(index as u32);
+            let core = CoreTextFixture::new(context.clone(), authorities.clone(), own_index);
+            cores.push(core);
+        }
+        cores
+    }
+
+    pub(crate) async fn receive<T: Copy>(timeout: Duration, mut receiver: watch::Receiver<T>) -> T {
+        tokio::time::timeout(timeout, receiver.changed())
+            .await
+            .expect("Timeout while waiting to read from receiver")
+            .expect("Signal receive channel shouldn't be closed");
+        *receiver.borrow_and_update()
+    }
+
+    pub(crate) struct CoreTextFixture {
```

Comment:
Very nice refactor! I think `CoreTextFixture` and `create_cores()` can be pulled out of the `mod test` and put directly under `core`.

Reply:
I am using some test only references and so I will keep it here for now and see if I can remove those references in the future when I use the core test fixture in the randomized test.

Reply:
Not sure I understand. Does adding `#[cfg(test)]` under `core` help?